docs/install.md

```diff
@@ -1302,6 +1314,140 @@ deployment is in the `Failed` state, then one can look into the event(s) using
 > [`DeploymentSpec`](#deployment-crd-api).
 
 
+## PMEM-CSI operator API conversion webhook
+
+The operator release v0.9.0 deprecates the `v1alpha1` [deployment API](#pmem-csi-deployment-crd)
+in favor of `v1beta1`. To keep supporting a custom resource that uses the
+`v1alpha1` version must be converted to the latest supported `v1beta1` CRD
+version. This conversion is done by the Kubernetes API server by using the
+conversion strategy chosen in the [deployment CRD sepcification](#pmem-csi-deployment-crd).
+If no conversion strategy is chosen by the CRD, by default the API server
+converts the custom resource `apiVersion` to the latest supported CRD version.
+
+This default conversion works as long as the deployment custom resource has no
+fields that were deprecated in the latest version. Such as `v1beta1` deprecates
+`nodeResources` and `controllerResources` in favor of per-container resource
+requests. These fields are silently ignored.
+
+```console
+$ kubectl get crd deployments.pmem-csi.intel.com -o jsonpath='{.spec.conversion}{"\n"}'
+{"strategy":"None"}
+
+$ kubectl create -f - << EOF
+apiVersion: pmem-csi.intel.com/v1alpha1
+kind: Deployment
+metadata:
+  name: pmem-csi.intel.com
+spec:
+  deviceMode: lvm
+EOF
+
+deployment.pmem-csi.intel.com/pmem-csi.intel.com created
+
+$ kubectl get deployments.pmem-csi.intel.com pmem-csi.intel.com -o jsonpath='{.apiVersion}{"\n"}{.spec}{"\n"}'
+pmem-csi.intel.com/v1beta1
+{"deviceMode":"lvm"}
+
+$ kubectl create -f - << EOF
+apiVersion: pmem-csi.intel.com/v1alpha1
+kind: Deployment
+metadata:
+  name: pmem-csi.intel.com
+spec:
+  deviceMode: lvm
+  nodeResources:
+    limits:
+      cpu: "500m"
+      memory: "250Mi"
+  controllerResources:
+    limits:
+      cpu: "400m"
+      memory: "150Mi"
+EOF
+
+deployment.pmem-csi.intel.com/pmem-csi.intel.com created
+
+$ kubectl get deployments.pmem-csi.intel.com pmem-csi.intel.com -o jsonpath='{.apiVersion}{"\n"}{.spec}{"\n"}'
+pmem-csi.intel.com/v1beta1
+{"deviceMode":"lvm"}
+```
```

Comment:
It's worth showing here what happens when a client asks for a v1alpha1 version of that same object.

Reply:
> It's worth showing here what happens when a client asks for a v1alpha1 version of that same object.

It supposed to return the same object by setting `apiVersion` to `v1alpha1`

Reply:
But with `controllerResources` and `nodeResources` set, so not quite the same. This illustrates what we need to mention earlier about objects being converted on demand instead of being rewritten in the etcd store.

Reply:
Long-term that becomes relevant for admins: if we ever *remove* the v1alpha1 API, they need to ensure that none of the stored CRs use that version or they'll become unusable.

Reply:
> But with `controllerResources` and `nodeResources` set, so not quite the same. This illustrates what we need to mention earlier about objects being converted on demand instead of being rewritten in the etcd store.

Only the latest(`v1beta1`)  CR fields get stored and when the client asks for a different revision, then the conversion happens from the stored revision. In this case where no conversion webhook, the `nodeResource`, `controllerResources` fields are unknown to `v1beta1` get lost and hence no change if we ask for `v1beta1` or `v1alpha1`.